chameleon/src/main/java/org/hibernate/omm/jdbc/MongoStatement.java

```diff
@@ -58,57 +42,53 @@ public MongoStatement(MongoDatabase mongoDatabase, ClientSession clientSession,
         this.connection = connection;
     }
 
-    private Document runCommand(Document command) {
-        return mongoDatabase.runCommand(clientSession, command);
-    }
-
     @Override
     public ResultSet executeQuery(String sql) throws SimulatedSQLException {
         Assertions.notNull("sql", sql);
         throwExceptionIfClosed();
-        Document command = Document.parse(sql);
-        Document commandResult = runCommand(command);
-        if (commandResult.getDouble(OK_FIELD) != 1.0) {
-            throw new CommandRunFailSQLException(commandResult);
-        }
-        return new MongoResultSet(commandResult);
+        BsonDocument command = BsonDocument.parse(sql);
+        MongoCollection<BsonDocument> collection = mongoDatabase.getCollection(command.getString("aggregate").getValue(),
+                BsonDocument.class);
+        List<BsonDocument> pipeline = command.getArray("pipeline").stream().map(BsonValue::asDocument).toList();
+        MongoCursor<BsonDocument> cursor = collection.aggregate(clientSession, pipeline).cursor();
+
+        return new MongoResultSet(cursor,
+                pipeline.get(pipeline.size() - 1).asDocument().getDocument("$project").keySet().stream().toList());
```

Comment:
I'm not entirely sure this is ok, but it seems to me that the last stage will/should always be a $project, in which case we can just get the field names from that stage instead of from the (possible empty) result set.

Reply:
There are multiple subtletiness here, including:

- if result set is empty, usually we don't need to bother with field names in the first place, so to me old way is enough
- do we need to consider the subtlety that `_id` field will be included implicitly if not specified? Then it involves nontrivial logic here (as I've encounntered)
- Hibernate ignored field name totally since v6 and only relies on field position index; for that reason, in my table joining simulation logic, I put ad-hoc field name there (only to ensure order or position index). So the above change will break another testing case and basically conflicts with my existing code logic to generate the `$project` field in `MongoSelectQueryAstTranslator` as below:
```
if (sqlSelection.getExpression() instanceof ColumnReference columnReference) {
                    appendSql("f" + i); // field name doesn't matter for Hibernate ResultSet retrieval only relies on order since v6
                    appendSql(": \"$");
                    appendSql(pathTracker.renderColumnReference(columnReference));
                    appendSql('"');
                } else {
```

When we take into consideration of table joinining, it is not easy to map projection field to entity field for it involves multiple entities and even in SQL each field is prefixed with table alias.

Reply:
btw, Hibernate v6 initially put lots of perf improvement hope on replacing ResultSet field retrieval by name with by position. It involves big change in code logic and the final benchmark shows almost the same perf. However, its real benefit might lies elsewhere, like the resulting SQL's conciseness and maybe Chameleon could benefit by this (otherwise table joininig simulation would be much much more challenging).